core/src/test/java/org/bitcoinj/store/WalletProtobufSerializerTest.java

```diff
@@ -356,9 +356,9 @@ public void testRoundTripWatchingWallet() throws Exception {
         assertEquals(creationTime.getEpochSecond(), wallet.getWatchingKey().getCreationTimeSeconds());
         assertEquals(creationTime.getEpochSecond(), wallet2.getWatchingKey().getCreationTimeSeconds());
         assertEquals(creationTime.getEpochSecond(), wallet3.getWatchingKey().getCreationTimeSeconds());
-        assertEquals(creationTime.getEpochSecond(), wallet.getEarliestKeyCreationTime());
-        assertEquals(creationTime.getEpochSecond(), wallet2.getEarliestKeyCreationTime());
-        assertEquals(creationTime.getEpochSecond(), wallet3.getEarliestKeyCreationTime());
+        assertEquals(creationTime, wallet.getEarliestKeyCreationTimeInstant());
+        assertEquals(creationTime, wallet2.getEarliestKeyCreationTimeInstant());
+        assertEquals(creationTime, wallet3.getEarliestKeyCreationTimeInstant());
```

Comment:
IIRC, this was where I was seeing a failure due to mock clock only having seconds-level resolution.

Reply:
It hadn't been of a problem in this one, but in others. I currently use `.truncatedTo(ChronoUnit.SECONDS)` when that happens, but in this case it isn't necessary (Instant.ofEpochSecond() is always second precision).
 
In future, I plan to propose introducing the notion of a "Bitcoin time" for certain data structures (mainly P2P messages), and maybe checks at certain places that force users to provide instants without millis or nanos.